File: frontend/src/core/infrastructure/repositoryImpl/organization/permissionRepository.ts
import { IPermissionRepository } from "@/core/domain/repository/organization/IPermissionRepository";
import HttpClient from "../../http/client";
import { Permission } from "@/core/domain/models/organization/permission";

export class PermissionRepository implements IPermissionRepository {

  constructor(
    private httpClient: HttpClient
  ) {}

  getAll(): Promise<Permission[]> {
    return this.httpClient.get<Permission[]>(`/permissions/all`);
  }
}

File: backend/API/Infrastructure/RepositoryImpl/Organization/PermissionRepository.cs
using API.Domain.Model.Organization;
using API.Domain.Repository.Organization;
using API.Domain.Repository.System;
using API.Infrastructure.Common.RepositoryImpl;
using API.Infrastructure.Db;
using Microsoft.EntityFrameworkCore;

namespace API.Infrastructure.RepositoryImpl.Organization
{
    public class PermissionRepository : BaseRepositoryImpl<Permission>, IPermissionRepository
    {
        public PermissionRepository(
            ConnDbContext context,
            IAuditLogRepository auditLogRepository
        ) : base(context, auditLogRepository)
        {
        }

        public IQueryable<Role> QueryWithPermissions()
        {
            return Context.Roles
                .Include(r => r.RolePermissions)
                    .ThenInclude(rp => rp.Permission)
                .AsNoTracking();
        }
    }
}
